### PR TITLE
invalidate cache on workflow update - fixes #23

### LINKED ISSUE #23: Prevent cache failures on workflow update
I was implementing the `transform` function of the `alfy.fetch` method in `alfred-ng2` and noticed a subtle area that could cause your workflow from failing a certain amount of time due to the cache being invalid.

Let me give you an example first to make it easier to understand. Let's assume I was calling an XML API that I transformed to JSON. At first (pre-transform), it would look like this.

```js
alfy.fetch(`https://my-api.com/resources.xml`, {maxAge: ONE_WEEK})
    .then(result => xml2js(result))
    .then(jsonResult => {
         // transform the jsonResult to alfred items
    });
```

The *XML* data is cached for one day. Now let's assume I upgrade to the transform function. This could be beneficiel because it only has to run the `xml2js` call once before caching.

```js
alfy.fetch(`https://my-api.com/resources.xml`, {
    maxAge: ONE_WEEK,
    transform: result => xml2js(result)
})
.then(jsonResult => {
    // transform the jsonResult to alfred items
});
```

An important thing to notice here is that in the second example, not the XML result is cached, but the result of the transform function and thus the JSON result is being cached. Pretty straightforward.

BUT, without knowing it, we could break everyones workflow now for one day. The thing is that if I had used this workflow quite recently, the XML data would still be cached and the `fetch` promise will not return JSON but XML. Boom, workflow explodes! Adding a function to (`transform` in this case), doesn't change anything to the [cache key](https://github.com/sindresorhus/alfy/blob/master/index.js#L124).

How did I solve this? By adding a `version` property to the options object.

```js
alfy.fetch(`https://my-api.com/resources.xml`, {
    maxAge: ONE_WEEK,
    version: pkg.version,
    transform: result => xml2js(result)
})
.then(jsonResult => {
    // transform the jsonResult to alfred items
});
```

This will make sure that my cache key is different on every release and will invalidate the old cache.

Is this something we could add in `alfy`? Could we detect the callers package.json version number? In my case `alfred-ng2` so that the developer doesn't have to worry about this issue? I'm quite sure that this issue will be overlooked by almost everyone if it was not tested properly.

A downside of this is that the cache could grow quite rapidly depending on the release cycles of the workflow.

### PR DESCRIPTION
**Pull request by SamVerschueren**
This PR fixes #23.

I also made sure that when requiring `alfy` in the tests, it uses a fresh version so it will use a different cache for each test.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -123,7 +123,9 @@ alfy.fetch = (url, opts) => {
 
 	const rawKey = url + JSON.stringify(opts);
 	const key = rawKey.replace(/\./g, '\\.');
-	const cachedResponse = alfy.cache.store[rawKey] && alfy.cache.store[rawKey].data;
+
+	const cachedData = (alfy.cache.store[rawKey] && alfy.cache.store[rawKey].data) || {};
+	const cachedResponse = cachedData.version === alfy.meta.version ? cachedData.data : undefined;
```
Comment (SamVerschueren) [inline index.js]: I don't think we should use `semver` here, right? It has to be an exact match imo.

File: package.json [hunk 2]
```diff
@@ -54,6 +54,7 @@
     "ava": "*",
     "delay": "^1.3.1",
     "nock": "^8.0.0",
+    "tempfile": "^1.1.1",
```
Comment (SamVerschueren) [inline package.json]: On mobile now but I believe this can be removed.

**Comment by SamVerschueren** [comment]
Don't merge yet. Have to check some things first (afk right now)

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js]: Yup, exact match.

**Comment by SamVerschueren** [comment]
Extracted the `alfy` requiring to an `utils.js` file to make it easier to write more independent tests in the future. Let me know if you want it differently (different name for instance).

File: test/cache.js [hunk 3]
```diff
@@ -1,21 +1,19 @@
 import test from 'ava';
 import delay from 'delay';
+import {alfy} from './fixtures/utils';
```
Comment (sindresorhus) [inline test/cache.js]: It's not really a fixture though. I would name it `'./fixtures/utils'` => `'./_util'`

Comment (SamVerschueren) [inline test/cache.js]: Definitely better. Put it in there to make sure AVA didn't process it, forgot that `_` prevents this as well.

**Pull request merged**